bench-vortex/src/public_bi_data.rs

```diff
@@ -0,0 +1,500 @@
+            (TrainsUK2, vec![
+                PBIUrl::new("TrainsUK2", "TrainsUK2_1.csv.bz2"),
+                PBIUrl::new("TrainsUK2", "TrainsUK2_2.csv.bz2")]),
+            (USCensus, vec![
+                PBIUrl::new("USCensus", "USCensus_1.csv.bz2"),
+                PBIUrl::new("USCensus", "USCensus_2.csv.bz2"),
+                PBIUrl::new("USCensus", "USCensus_3.csv.bz2")]),
+            (Uberlandia, vec![
+                PBIUrl::new("Uberlandia", "Uberlandia_1.csv.bz2")]),
+            (Wins, vec![
+                PBIUrl::new("Wins", "Wins_1.csv.bz2"),
+                PBIUrl::new("Wins", "Wins_2.csv.bz2"),
+                PBIUrl::new("Wins", "Wins_3.csv.bz2"),
+                PBIUrl::new("Wins", "Wins_4.csv.bz2")]),
+            (YaleLanguages, vec![
+                PBIUrl::new("YaleLanguages", "YaleLanguages_1.csv.bz2"),
+                PBIUrl::new("YaleLanguages", "YaleLanguages_2.csv.bz2"),
+                PBIUrl::new("YaleLanguages", "YaleLanguages_3.csv.bz2"),
+                PBIUrl::new("YaleLanguages", "YaleLanguages_4.csv.bz2"),
+                PBIUrl::new("YaleLanguages", "YaleLanguages_5.csv.bz2")]),
+        ]);
+}
+
+impl PBIDataset {
+    pub fn dataset_name(&self) -> String {
+        let url = URLS.get(self).unwrap();
+        url.first().unwrap().dataset_name.clone()
+    }
+
+    pub fn fname_from_url(url: String) -> String {
+        url.split('/').last().unwrap().to_string()
+    }
+
+    fn csv_files(&self) -> Vec<PathBuf> {
+        let urls = URLS.get(self).unwrap();
+        self.dataset_name();
+        urls.iter().map(|url| self.get_csv_path(url)).collect_vec()
+    }
+
+    fn get_csv_path(&self, url: &PBIUrl) -> PathBuf {
+        data_path("PBI")
+            .join(self.dataset_name())
+            .join("csv")
+            .join(url.file_name.strip_suffix(".bz2").unwrap())
+    }
+
+    fn download_bzip(&self) {
+        let urls = URLS.get(self).unwrap();
+        self.dataset_name();
+        urls.iter().for_each(|url| {
+            let fname = self.get_bzip_path(url);
+            download_data(fname, url.to_url_string().as_str());
+        });
+    }
+
+    fn get_bzip_path(&self, url: &PBIUrl) -> PathBuf {
+        data_path("PBI")
+            .join(self.dataset_name())
+            .join("bzip2")
+            .join(url.file_name.clone())
+    }
+
+    fn unzip(&self) {
+        for url in URLS.get(self).unwrap() {
+            let bzipped = self.get_bzip_path(url);
+            let unzipped_csv = self.get_csv_path(url);
+            decompress_bz2(bzipped, unzipped_csv);
+        }
+    }
+}
+
+#[derive(Debug)]
+struct PBIUrl {
+    dataset_name: String,
+    file_name: String,
+}
+
+impl PBIUrl {
+    fn new(dataset_name: &str, file_name: &str) -> Self {
+        PBIUrl {
+            dataset_name: dataset_name.to_string(),
+            file_name: file_name.to_string(),
+        }
+    }
+    fn to_url_string(&self) -> String {
```

Comment:
you have a bunch of useful functionality in https://docs.rs/reqwest/latest/reqwest/struct.Url.html that can simplify parsing and formatting